### ISSUE TITLE
Custom Torch Model Support

### ISSUE THREAD

**Issue by BiEchi** [neelnanda-io/TransformerLens#498]
Hi @neelnanda-io , is there any support/tutorials on applying this repo to custom models on pytorch (not based on HF)?

**Comment by neelnanda-io** [comment]
No current tutorials, happy to give advice, but no support beyond that.
What do you want to do?

**Comment by BiEchi** [comment]
I've got a custom model based on [nanogpt](https://github.com/karpathy/nanoGPT). I simply changed several lines of code and then pretrained from scratch. As lots of mech interp tools are based on TransformerLens, I'm interested in applying the model to Transformer Lens.

**Comment by neelnanda-io** [comment]
Yes, that works fine. The procedure is to manually define the config to
create a blank HookedTransformer, then use
loading_from_pretrained.convert_nanogpt_weights to convert the
nanogpt model weights to transformerlens format, then load them in. See
this notebook from @adamkarvonen for a demo (note
that convert_nanogpt_weights is now a TransformerLens function, thanks to
Adam!)
https://github.com/adamkarvonen/chess_llm_interpretability/blob/main/nanogpt_to_transformer_lens.ipynb

**Comment by BiEchi** [comment]
Thanks a lot for your response @neelnanda-io 
My new model has some tweaks to the model architecture, so it's not directly transferable. Would you like to point out what parts I need to modify if I want to modify the model arch?

**Comment by BiEchi** [comment]
@neelnanda-io I've successfully set up Nanogpt weight transfer, but I don't see the interface for calling a GPT model from Nanogpt repos. I.e., I don't see the function `convert_nanogpt_weights` called anywhere in the repo. Is it supported or we need to write it ourselves?

**Comment by adamkarvonen** [comment]
This may help: https://github.com/neelnanda-io/TransformerLens/blob/829084a53836c5b8b388aa37a5ffce73b6371712/transformer_lens/loading_from_pretrained.py#L1205

It's just one additional line to call `convert_mingpt_weights`.

**Comment by BiEchi** [comment]
Thanks @neelnanda-io @adamkarvonen , the problem is perfectly solved.
